### PR TITLE
Fix adminUpdate steps

### PR DESCRIPTION
**Pull request by mociarain**
### Which issue this PR addresses:

A refactor to lay the groundwork for: https://issues.redhat.com/browse/ARO-5327

### What this PR does / why we need it:

I've smashed all the various `if` blocks to give us another view on what this is doing and hopefully, this is clearer. I've preserved the step execution order (mostly) because it's not clear to me what exactly what dependencies are involved.   Functionally, I believe this is identical to what we had before but now there is some extra repeats of certain idempotent steps. 

Note: I noticed that in the `everything` path weren't actually running the `m.renewMDSDCertificate` step. This PR fixes this.

Some questions:

- ~~Is it valid to run this with all flags set to false? Technically, it would have done bootstrap 0 & 1 and step 0 but here I set it to panic.~~
It is valid according to the unit tests.

- ~~Is the hive stuff dependant on `initializeOperatorDeployer`?~~
No.

Related PUCM scipt PR: https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/ARO-Scripts/pullrequest/9515547

### Test plan for issue:

I've tested this and it's using my local RP to prove that:
- [x] The everything step runs everything happily.
- [x] The individual "branches" can be ran successfully.

### Is there any documentation that needs to be updated for this PR?

https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/WIKI-Designs/pullrequest/9515543

### Future Ideas/Thoughts:

1. Don't grow `GeneralFixesSteps`
This is now a kind of dumping ground of random things that we should stop adding to IMO. It's only going to snowball and hopefully what I have done here shows how we could define extra groups and add them with correct context.
1. Break up  `GeneralFixesSteps`
This would involve understanding the dependencies between steps and more understanding than me to group them better conceptually.

### CODE REVIEW

**Comment by mociarain** [comment]
This is an un-forked version of an existing PR: https://github.com/Azure/ARO-RP/pull/3376

**Review by kimorris27 — CHANGES_REQUESTED** [review]
This is a good improvement. I found one small regression and recommended a way to fix it, but LGTM otherwise.

File: pkg/cluster/install.go [hunk 1]
```diff
@@ -57,109 +89,73 @@ func (m *manager) adminUpdate() []steps.Step {
 		// struct, so we'll rebuild the fpAuthorizer and use the error catching
 		// to advance
 		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.fixupClusterSPObjectID),
-		steps.Action(m.fixInfraID), // Old clusters lacks infraID in the database. Which makes code prone to errors.
+		steps.Action(m.startVMs),
+		steps.Condition(m.apiServersReady, 30*time.Minute, true),
```
[the remark this answers is not in the document]
Comment (mociarain) [reply pkg/cluster/install.go:93]: Hmmm. I think I agree with you. I'll patch that now

[the remark this answers is not in the document]
Comment (mociarain) [reply pkg/cluster/install.go]: Done

**Review by kimorris27 — APPROVED** [review]

**Review by bitoku — APPROVED** [review]
slices.Concat was added from 1.22, unfortunately we can't use it yet.
https://pkg.go.dev/slices#Concat

**Review by SrinivasAtmakuri — APPROVED** [review]
LGTM!

**Review by tiguelu — APPROVED** [review]
lgtm

**Review by tiguelu — APPROVED** [review]

**Pull request merged**